assembly/src/docs/20_INSTALL_WINDOWS_WITH_GRAALVM.md

```diff
@@ -0,0 +1,133 @@
+# Build TornadoVM with Graal 20.2.0 JDK 8 / 11 on Windows x64
+## Install tools
+### Maven
+Download Apache Maven from the [official site](https://maven.apache.org/download.cgi) and extract it to the any location. Below it's assumed that Maven's home is `C:/Maven`.
+### MSys2
+1. Download [MSys2](https://www.msys2.org/) installer from the official site and run it. You may choose any installation directory, below it will be referred as `<MSYS2>`. IMPORTANT: the only executable you should use as a terminal is `<MSYS2>/mingw64.exe`, don't use others (msys2.exe, mingw32.exe) as this would lead to unpredictable results.
+
+2. Update MSys2 _**system**_ packages. Start `<MSYS2>/mingw64.exe` and run the following command in the terminal:
+```bash
+pacman -Syu
+```
+You might need to execute it several times until you see that no updates found.
+
+3. Update MSys2 _**default**_ packages. In the terminal window of `<MSYS2>/mingw64.exe` run:
+```bash
+pacman -Su
+```
+You might need to execute it several times until you see that no updates found.
+
+4. Install necessary tools to MSys2. In the terminal window of `<MSYS2>/mingw64.exe` run:
+```bash
+pacman -S \
+mingw-w64-x86_64-make           \
+mingw-w64-x86_64-cmake          \
+mingw-w64-x86_64-gcc            \
+mingw-w64-x86_64-opencl-headers \
+mingw-w64-x86_64-opencl-icd-git \
+python make git
+```
+
+5. Create default Maven repository for MSys2 user:
+```bash
+cd ~
+mkdir .m2
+```
+
+6. Create file `mvn` in `<MSYS2>/mingw64/bin` with any textual editor according to the template below:
+```bash
+#!/usr/bin/env bash
+BASH_FILE=$(where bash)
+SETTINGS=\"${BASH_FILE%\\*\\*\\*}${HOME}/.m2/settings.xml\"
+
+C:/Maven/bin/mvn.cmd --settings ${SETTINGS} "$@"
+```
+You should only change path to Maven executable to the actual location of Maven in your system.
+
+## Get TornadoVM sources
+Clone TornadoVM 0.8 from Git repository using `<MSYS2>/mingw64.exe`:
+```
+cd D:/MyProjects
+git clone https://github.com/beehive-lab/TornadoVM.git
+cd TornadoVM
+```
+We will refer hereafter the directory with TornadoVM sources as `<TornadoVM>`.
+
+## Get Graal 20.2.0
+Visist [GraalVM Community Edition 20.2.0](https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-20.2.0) and download either of the following builds:
+- [graalvm-ce-java8-windows-amd64-20.2.0.zip](https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-20.2.0/graalvm-ce-java8-windows-amd64-20.2.0.zip) -- Graal JDK 8 x64
+- [graalvm-ce-java11-windows-amd64-20.2.0.zip](https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-20.2.0/graalvm-ce-java11-windows-amd64-20.2.0.zip) -- Graal JDK 11 x64
+
+Extract the downloaded file to any directory.
+
+## Install NVIDIA CUDA and NVIDIA driver
```

Comment:
I believe there are a few options. What TornadoVM needs is either
a) an OpenCL implementation (e.g. from CUDA, Intel OpenCL, AMD OpenCL)
b) a PTX implementation (CUDA) 

So it is possible to use, for example, Intel OpenCL instead of CUDA. I would reflect this in this document.

Reply:
It's reflected right after title